openfisca_aotearoa/variables/acts/social_security/schedule_4/part3.py

```diff
@@ -0,0 +1,324 @@
+
+    def formula_2018_11_26(persons, period, parameters):
+        single = numpy.logical_not(persons("social_security__in_a_relationship", period))
+        no_children = persons("social_security__dependent_children", period.first_week) == 0
+        not_1a = numpy.logical_not(persons("schedule_4__part3_1_a", period))
+        return single * no_children * not_1a
+
+
+class schedule_4__part3_1_c(variables.Variable):
+    value_type = bool
+    entity = entities.Person
+    definition_period = periods.WEEK
+    reference = "https://legislation.govt.nz/act/public/2018/0032/latest/DLM6784861.html"
+    label = "Part 3 Supported Living Payment - Clause 1(c)"
+
+    def formula_2018_11_26(persons, period, parameters):
+        single = numpy.logical_not(persons("social_security__in_a_relationship", period))
+        has_children = persons("social_security__dependent_children", period.first_week) > 0
+        return single * has_children
+
+
+class schedule_4__part3_1_d(variables.Variable):
+    value_type = bool
+    entity = entities.Person
+    definition_period = periods.WEEK
+    reference = "https://www.legislation.govt.nz/act/public/2018/0032/latest/DLM6784861.html"
+    label = "Part 3 Supported Living Payment - Clause 1(d)"
+
+    def formula_2018_11_26(people, period, parameters):
+        in_relationship = people("social_security__in_a_relationship", period)
+        partner_granted_main_benefit = people.family.any(
+            people.family.members("social_security__granted_main_benefit", period),
+            role=entities.Family.PARTNER)
+        return in_relationship * partner_granted_main_benefit
+
+
+class schedule_4__part3_1_d_i(variables.Variable):
+    value_type = bool
+    entity = entities.Person
+    definition_period = periods.WEEK
+    reference = "https://www.legislation.govt.nz/act/public/2018/0032/latest/DLM6784861.html"
+    label = "Part 3 Supported Living Payment - Clause 1(d)(i)"
+
+    def formula_2018_11_26(people, period, parameters):
+        base = people("schedule_4__part3_1_d", period)
+        no_children = people("social_security__dependent_children", period) < 1
+        return base * no_children
+
+
+class schedule_4__part3_1_d_ii(variables.Variable):
+    value_type = bool
+    entity = entities.Person
+    definition_period = periods.WEEK
+    reference = "https://www.legislation.govt.nz/act/public/2018/0032/latest/DLM6784861.html"
+    label = "Part 3 Supported Living Payment - Clause 1(d)(ii)"
+
+    def formula_2018_11_26(people, period, parameters):
+        base = people("schedule_4__part3_1_d", period)
+        has_children = people("social_security__dependent_children", period) < 1
+        return base * has_children
+
+
+class schedule_4__part3_1_e(variables.Variable):
+    value_type = bool
+    entity = entities.Person
+    definition_period = periods.WEEK
+    reference = "https://www.legislation.govt.nz/act/public/2018/0032/latest/DLM6784861.html"
```

Comment:
I'd suggest we also include a reference to the original version of the 2018 act so it's easier to track down the original formula and perhaps a link as well to the date it changed:
https://www.legislation.govt.nz/act/public/2018/0032/77.0/DLM6784861.html
https://www.legislation.govt.nz/act/public/2018/0032/140.0/DLM6784861.html